lib/licensed/sources/swift.rb

```diff
@@ -7,15 +7,16 @@ module Licensed
   module Sources
     class Swift < Source
       def enabled?
-        return unless Licensed::Shell.tool_available?("swift") && swift_package?
+        return unless Licensed::Shell.tool_available?("xcodebuild")
+        @derived_data_path = derived_data_path
```

Comment:
```suggestion
```

Reply:
Je ne sais pas ce que tu en penses : ma décision de mettre le résultat de derived_data_path dans une variable, c'est que la commande xcodebuild est vraiment lente à exécuter. 

J'ai l'impression que si on call directement la méthode derived_data_path dans dependency_path_for_url ça va être vraiment vraiment slow de caller xcodebuild à chaque itération de la loop.

Reply:
Si tu vas de l'avant avec cette [proposition](https://github.com/agendrix/licensed/pull/1#discussion_r1204202569), elle va être dans une variable 😄

Reply:
Ahhh bien oui tu as bien raison. Je suis gelé 😬

Reply:
Oui 🥲 

```
def derived_data_path
 @derived_data_path = JSON.parse(`xcodebuild -showBuildSettings $@ -json`).first.dig("buildSettings", "BUILD_DIR").delete_suffix("/Build/Products")
end

Reply:
Ok! Je ne sais pas si c'est ruby-way, mais dans ce cas, j'aurais suggéré de caller la méthode `set_derived_data_path` et la caller dans le enabled?

```ruby
      def enabled?
        return unless Licensed::Shell.tool_available?("xcodebuild")
        set_derived_data_path
        File.exist?(package_resolved_file_path)
      end

      def set_derived_data_path
        build_dir = JSON.parse(`xcodebuild -showBuildSettings $@ -json`).first.dig("buildSettings", "BUILD_DIR")
        @derived_data_path ||= build_dir.delete_suffix("/Build/Products")
      end
```

Reply:
C'est sûr que techniquement parlant, enabled ne devrait peut-être pas avoir la responsabilité de setter le path...mais bon

Reply:
Personnellement j'aime pas set un path dans une méthode qui retourne un booléen, mais as you wish, tant que ca roule la commande à chaque fois ca me va!